### DISCUSSION TITLE
VsCode extension

### DISCUSSION THREAD

**Discussion by robert-dean** [tareqimbasher/NetPad#136] [General]
Would there be interest in potentially splitting out the UI and having the core as a stand-alone library which could then be wrapped as an extension in vscode?

**Comment by rossknudsen** [comment]
Yeah I think that would be an interesting idea and would be interested in working on it.  Just need a bit better understanding of the current project structure to do some experimentation.

**Comment by rossknudsen** [comment]
https://yuml.me/edit/3ecf814f

**Comment by rossknudsen** [comment]
Reviewing existing functionality, a lot can be ignored and left up to VS Code.  This is what I think needs to be ported over:

Settings:

- SDKs (select folders/automatic selection)
- Scripts location
- Package Cache location
- Open when script runs
- Text wrap
- Serialization - max depth
- Max Collection Length
- Omnisharp settings...

Editor Capabilities:

- namespace configuration
- Nuget/dll references
- SDK selection
- file type (program, statements, expression)
- DB connection configuration
- save

**Comment by rossknudsen** [comment]
Reviewing different extension options:

Language server - but we want to leverage the existing C#/.Net capability, not reinvent the whole thing.
CustomTextEditor - not suitable because we have to render our own content, not reuse the existing editor
Virtual documents - not suitable since they seem to be readonly.
Virtual file system - this seems to be the most reasonable.
Virtual workspace - seems overkill as it will mean more configuration on our part.
